### PR TITLE
fix(shared-data): update flex fixed-trash for dvt

### PR DESCRIPTION
**Pull request by sfoster1**
This fixed trash bin has changed shapes.
## Review requests
- is it, like, ok that the position is negative? The depth really does extend below the deck

## testing
- should drop stuff a lot closer to the trash now

### CODE REVIEW

File: shared-data/labware/definitions/2/opentrons_1_trash_3200ml_fixed/1.json [hunk 1]
```diff
@@ -27,10 +27,10 @@
       "yDimension": 165.67,
       "xDimension": 107.11,
       "totalLiquidVolume": 3200000,
-      "depth": 187.5,
+      "depth": 172.5,
```
Comment (sanni-t) [inline shared-data/labware/definitions/2/opentrons_1_trash_3200ml_fixed/1.json]: The trash z dimension is 15mm more than the depth, so I'm assuming that's the thickness of the trash bottom material. Is that correct?

File: shared-data/labware/definitions/2/opentrons_1_trash_3200ml_fixed/1.json [hunk 2]
```diff
@@ -27,10 +27,10 @@
...
       "x": 105.375,
       "y": 39.875,
-      "z": 0
+      "z": -132.5
```
Comment (sanni-t) [inline shared-data/labware/definitions/2/opentrons_1_trash_3200ml_fixed/1.json]: So this means the trash goes 132.5mm below the deck, ya? But the `cornerOffsetFromSlot` has a z offset of -94.. I think these two should be the same so not sure which one needs to be corrected.

**Comment by SyntaxColoring** [comment]
> is it, like, ok that the position is negative? The depth really does extend below the deck

We should keep an eye out for motion planning accidentally clamping z-coordinates to 0.

We should also probably figure out a number for the lowest possible z that a pipette can move to, the same way we already have that for the highest possible z.

File: shared-data/labware/definitions/2/opentrons_1_trash_3200ml_fixed/1.json (hunk 2 above)
Comment (sfoster1) [reply shared-data/labware/definitions/2/opentrons_1_trash_3200ml_fixed/1.json]: oh I think I need to update the corner offset then

**Comment by sfoster1** [comment]
The negative positions fail protocols in simulation unfortunately. I'll rework this to just have the "above-ground" geometry.
Screenshot 2023-04-24 at 12 14 05 PM

**Comment by SyntaxColoring** [comment]
> @sfoster1 The negative positions fail protocols in simulation unfortunately.

It looks like that's stemming from the fact that the well `z` is restricted to being a `positiveNumber`:

https://github.com/Opentrons/opentrons/blob/970036b3dd0dd550920e400dde97995a397805be/shared-data/labware/schemas/2.json#L235-L236

Which makes sense.

> @sanni-t So this means the trash goes 132.5mm below the deck, ya? But the cornerOffsetFromSlot has a z offset of -94.. I think these two should be the same so not sure which one needs to be corrected.

Reading the schema docs more closely, I think `cornerOffsetFromSlot`'s `z` and the well's `z` should *not* be the same. My interpretation is:

* `cornerOffsetFromSlot` should have a very negative `z`. This reflects the fact that the bottom surface of the labware is below the surface of the slot.
* The well `z` should be a smallish positive number, to represent the thickness of the bin's plastic floor.

In other words, I think the well's `z` is already relative to the `cornerOffsetFromSlot`, so it doesn't make sense for it to be negative.

**Comment by sanni-t** [comment]
> In other words, I think the well's `z` is already relative to the `cornerOffsetFromSlot`, so it doesn't make sense for it to be negative.

Good catch! This is correct. Verified by calculating position of top of the trash well the way engine calculates it and it works correctly for `cornerOffsetFromSlot` of (x1, y1, -134.5), well depth of 172.5 and well z of 2.5 (thickness of the trash bottom).

**Comment by sfoster1** [comment]
That makes sense, thank you! I think though that we need changes like you mentioned @SyntaxColoring about minimum z positions before we allow a negative position that large, just in case someone tries to go to `.bottom()` or `.center()` or whatever. I think we should merge this with above-deck-geometry only for now.

**Review by sanni-t — APPROVED** [review]
Ideally I would like to get the true measurements for the trash bin in there.

That said, I don’t think we have any code in place that will prevent the pipette from trying to go to fixed_trash.bottom() which would be a problem so I can see the pros of using the above-deck-only approach. 

But I hope we do follow-up with true measurements + restricting the pipette from trying to reach the bin bottom (either by clipping the pipette travel distance to max allowed or by having the API raise a user-friendly error instead of relying on hardware control's axis limit errors) because you can still specify something like `well.bottom(-50)` with the current above-deck-only measurements and run into an error.

**Comment by sfoster1** [comment]
Definitely agree. And we want sub-deck stuff for calibration structures.
